pkg/image/admission/imagequalify/api/validation/validation.go

```diff
@@ -0,0 +1,28 @@
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/origin/pkg/image/admission/imagequalify/api"
+)
+
+func Validate(config *api.ImageQualifyConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if config == nil {
+		return allErrs
+	}
+	for i, rule := range config.Rules {
+		if rule.Pattern == "" {
```

Comment:
limit the charset in pattern to just characters valid in an image spec and *

Reply:
If we do checking in validation then by the time this is exercised only have valid patterns. Basically, you either have a correct config, or you don't. This is an internal call so validation should prevent you getting this far with a bad pattern.

Reply:
this is where you would do that validation

Reply:
Yep, agreed. I had your two comments the wrong way round in my head.

Reply:
> limit the charset in pattern to just characters valid in an image spec and *

Done.